functional.submit_and_retrieve_message AssertionError

```
test_submit_and_retrieve_happy_path (tests.functional.submit_and_retrieve_message.SubmitAndRetrieveMessage) ... FAIL

======================================================================
FAIL: test_submit_and_retrieve_happy_path (tests.functional.submit_and_retrieve_message.SubmitAndRetrieveMessage)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/functional/submit_and_retrieve_message.py", line 37, in test_submit_and_retrieve_happy_path
    self._journalist_downloads_message()
  File "tests/functional/journalist_navigation_steps.py", line 31, in _journalist_downloads_message
    self.assertEqual(1, len(submissions))
AssertionError: 1 != 0

----------------------------------------------------------------------
Ran 1 test in 15.706s

FAILED (failures=1)
```

Only happens in Vagrant, and not on Travis, which is... annoying.

--- a/securedrop/tests/functional/functional_test.py
+++ b/securedrop/tests/functional/functional_test.py
@@ -57,6 +57,10 @@ class FunctionalTest():
         self.journalist_process.start()
 
         self.driver = webdriver.Firefox()
+        # Poll the DOM briefly to wait for elements. It appears .click() does
+        # not always do a good job waiting for the page to load, or perhaps
+        # Firefox takes too long to render it (#399)
+        self.driver.implicitly_wait(1)
 
         self.secret_message = 'blah blah blah'
 

